src/solver/infeasible-problem-analysis/watched-constraints.cpp

```diff
@@ -154,22 +154,25 @@ std::string HydroProduction::infeasibilityCause()
 // --- Constraints factory ---
 ConstraintsFactory::ConstraintsFactory()
 {
-    regex_to_ctypes_ = {
-      {"::hourly::", [](std::string name) { return std::make_shared<HourlyBC>(name); }},
-      {"::daily::", [](std::string name) { return std::make_shared<DailyBC>(name); }},
-      {"::weekly::", [](std::string name) { return std::make_shared<WeeklyBC>(name); }},
-      {"^FictiveLoads::", [](std::string name) { return std::make_shared<FictitiousLoad>(name); }},
-      {"^AreaHydroLevel::", [](std::string name) { return std::make_shared<HydroLevel>(name); }},
-      {"^Level::", [](std::string name) { return std::make_shared<STS>(name); }},
-      {"^HydroPower::", [](std::string name) { return std::make_shared<HydroProduction>(name); }}};
-}
-
-std::shared_ptr<WatchedConstraint> ConstraintsFactory::create(std::string name) const
-{
-    return std::ranges::find_if(regex_to_ctypes_,
-                                [&name](auto& pair)
-                                { return std::regex_search(name, std::regex(pair.first)); })
-      ->second(name);
+    regex_to_ctypes_ = {{"::hourly::", std::make_unique<HourlyBC, const std::string&>},
```

Comment:
ok, I did not realize that, regarding **std::make_unique**, one could add the target **T**'s constructor arguments' types as template parameters. Good to know.

Reply:
If you want the function pointer, it's mandatory since `std::make_unique` has 6 versions. The compiler can't determine what version you're calling in this context. Here we want the version that calls the constructor `HourlyBC::HourlyBC(const std::string&)`, so that's how we tell him.